File: rust/src/models/sub_line_item.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SubLineItem {
    pub id: String,

    pub name: String,

    pub quantity: rust_decimal::Decimal,

    pub total: i32,

    pub unit_price: rust_decimal::Decimal,
}

impl SubLineItem {
    pub fn new(
        id: String,
        name: String,
        quantity: rust_decimal::Decimal,
        total: i32,
        unit_price: rust_decimal::Decimal,
    ) -> Self {
        Self {
            id,
            name,
            quantity,
            total,
            unit_price,
        }
    }
}

File: typescript/src/models/subLineItem.ts
// this file is @generated

export interface SubLineItem {
  id: string;

  name: string;

  quantity: string;

  total: number;

  unitPrice: string;
}

export const SubLineItemSerializer = {
  _fromJsonObject(object: any): SubLineItem {
    return {
      id: object["id"],
      name: object["name"],
      quantity: object["quantity"],
      total: object["total"],
      unitPrice: object["unit_price"],
    };
  },

  _toJsonObject(self: SubLineItem): any {
    return {
      id: self.id,
      name: self.name,
      quantity: self.quantity,
      total: self.total,
      unit_price: self.unitPrice,
    };
  },
};
